specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,175 @@
+  tail call void @__quantum__rt__result_record_output(%Result* inttoptr (i64 1 to %Result*), i8* getelementptr inbounds ([3 x i8], [3 x i8]* @2, i32 0, i32 0))
+
+  ret i64 0
+}
+
+; declarations of QIS functions
+
+declare void @__quantum__qis__h__body(%Qubit*)
+
+declare void @__quantum__qis__cnot__body(%Qubit*, %Qubit*)
+
+declare void @__quantum__qis__mz__body(%Qubit*, %Result*)
+
+; declarations of functions used for output recording
+
+declare void @__quantum__rt__initialize_record_output(i8*)
+
+declare void @__quantum__rt__tuple_record_output(i64, i8*)
+
+declare void @__quantum__rt__result_record_output(%Result*, i8*)
+
+; attributes
+
+attributes #0 = { "entry_point" "required_qubits"="2" "required_results"="2" }
+```
+
+TODO: do we need additional restrictions for output recording? Like e.g. it needs to be a dedicated block, or output recording functions can only occur in the last block of the entry point?
+
+TODO: do we need a __quantum__rt__initialize and __quantum__rt__finalize?
+TODO: do measurements need to be at the end of the quantum computations, or should the backend take care of deferring them if necessary?
+
+TODO: a profile identifier and version number within the IR would be good to have (may be needed for correct usage of qubit and result pointers)   
+-> look into custom target triples
+
+## Entry Point Definition
+
+The bitcode contains the definition of the LLVM function that should be invoked when the program is executed, 
+refered to as entry point in the rest of this profile specification. 
+The name of this function may be chosen freely, as long as it is a valid 
+[global identifier](https://llvm.org/docs/LangRef.html#identifiers) by LLVM standard. 
+
+The entry point may not take any parameters and must return an exit code in the form of a 64-bit integer. The exit code 0 must be used to indicate a successful execution of the program. 
+
+### Attributes
+
+The following custom attributes must be attached to the entry point function:
+
+- An attribute named "EntryPoint" that does not 
+- An attribute indicating the total number of qubits used by the program
+- An attribute indicating the total number of classical bits required throughout the program to store measurement outcomes
+
+These attributes will show up as an [attribute group](https://releases.llvm.org/13.0.1/docs/LangRef.html#attrgrp) in the IR.
+Attribute groups are numbered such that they can be easily referenced by multiple function definitions or global variables. 
+Arbitrary custom attributes may be optionally attached to any of the declared functions to convey additional information about that function. Consumers of Base Profile compliant programs should hence not rely on the numbering of the entry point attribute group, but instead look for function to which an attribute with the name `entry_point` is attached to determine which one to invoke when the program is launched.
```

Comment:
Naming `entry_point` is inconsistent with previous section where it was `EntryPoint` and is also inconsistent with the current convention by Q# that uses `EntryPoint`.

Reply:
updated, with `entry_point` being the intended naming.